### PR TITLE
added new feature of email address suggestion

### PR DESCRIPTION
**Pull request by arjunvijayvargiya**
This feature will add some functionality to the log-in screen.It will suggest email addresses typed by the user earlier in his/her phone.So that user wont be typing it again.This will save some type from user perspective.
Apart from that it remembers the email id used by the user in his previous log-in.

### CODE REVIEW

File: src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java [hunk 1]
```diff
@@ -30,37 +35,62 @@
...
 import ly.priv.mobile.utils.Utilities;
 import ly.priv.mobile.utils.Values;
 
 public class LoginActivity extends Activity {
     ViewSwitcher switcher;
     TextView contentServerTextView;
-    EditText contentServerEditText, emailAddressEditText, pwdEditText;
+    EditText contentServerEditText, pwdEditText;
+    AutoCompleteTextView emailAddressEditText;
+    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,4}$", Pattern.CASE_INSENSITIVE);
```
Comment (vshivam) [inline src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java]: This is a constant value and you should move it to the /ly/priv/mobile/utils/ConstantValues.java class.

File: src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java [hunk 2]
```diff
@@ -30,37 +35,62 @@
...
     ImageButton saveContentServerButton;
     ProgressBar progressBar;
     Button loginButton;
     private Values mValues;
     String mContentServerDomain, authToken;
+    SharedPreferences prefs;
```
Comment (vshivam) [inline src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java]: The ly/priv/mobile/utils/Values.java class has a shared preferences file already defined. You should create new getters and setters in that class whenever you want to use shared prefs.

File: src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java [hunk 3]
```diff
@@ -30,37 +35,62 @@
...
+        prefs = getSharedPreferences(prefName, MODE_PRIVATE);
         switcher = (ViewSwitcher) findViewById(R.id.content_server_switcher);
         contentServerTextView = (TextView) findViewById(R.id.content_server_view);
         saveContentServerButton = (ImageButton) findViewById(R.id.save_content_server);
         contentServerEditText = (EditText) findViewById(R.id.content_server_edit_text);
         progressBar = (ProgressBar) findViewById(R.id.progressBar);
         loginButton = (Button) findViewById(R.id.btn_login);
-        emailAddressEditText = (EditText) findViewById(R.id.email_edit_text);
+        emailAddressEditText = (AutoCompleteTextView) findViewById(R.id.email_edit_text);
+        Account[] accounts = AccountManager.get(this).getAccounts();
```
Comment (vshivam) [inline src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java]: Line numbers 80 - 92 can be moved to a new method inside the ly/priv/mobile/utils/Utilities.java class.

File: src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java [hunk 4]
```diff
@@ -30,37 +35,62 @@
...
+        Set<String> emailSet = new HashSet<String>();
+        if(readPrefValues())
+        {
```
Comment (vshivam) [inline src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java]: Please make sure that the code formatting is constant. Curly braces go in the same line as the beginning of the if block. If you're using android studio, CTRL + ALT + L (Win, Ubuntu) or OPTION + CMD + L (Mac) to format the code.

File: src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java [hunk 5]
```diff
@@ -204,6 +242,10 @@ protected void onPostExecute(String mLoginResponse) {
                         String authToken = jObject.getString("auth_key");
                         mValues.setAuthToken(authToken);
                         mValues.setUserVerifiedAtLogin(true);
+                        SharedPreferences.Editor editor = prefs.edit();
```
Comment (vshivam) [inline src/main/java/ly/priv/mobile/gui/activities/LoginActivity.java]: All of this should go to a method inside the ly/priv/mobile/utils/Values.java class.

**Comment by arjunvijayvargiya** [comment]
I will make all the necessary changes suggested by you and will get back to you

**Pull request closed without merging**